lecture_3/03_project/calculator/internal/infrastructure/sqlite/betRepository.go

```diff
@@ -0,0 +1,186 @@
+		return errors.Wrap(err, "calc_bets repository failed to insert a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryInsertBet(ctx context.Context, bet storagemodels.BetCalculated) error {
+	// If payout is 0, do not insert it.
+	if bet.Payment == 0 {
+		insertBetSQL := "INSERT INTO bets(id, selection_id, selection_coefficient, payment) VALUES (?, ?, ?, ?)"
+		statement, err := r.dbExecutor.PrepareContext(ctx, insertBetSQL)
+		if err != nil {
+			return err
+		}
+
+		_, err = statement.ExecContext(ctx, bet.Id, bet.SelectionId, bet.SelectionCoefficient, bet.Payment)
+		return err
+	}
+
+	insertBetSQL := "INSERT INTO bets(id, selection_id, selection_coefficient, payment) VALUES (?, ?, ?, ?)"
+	statement, err := r.dbExecutor.PrepareContext(ctx, insertBetSQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = statement.ExecContext(ctx, bet.Id, bet.SelectionId, bet.SelectionCoefficient, bet.Payment)
+	return err
+}
+
+// UpdateBetCalculated updates the provided calculated bet in the database. An error is returned if the operation
+// has failed.
+func (r *BetRepository) UpdateBetCalculated(ctx context.Context, bet domainmodels.BetCalculated) error {
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryUpdateBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "calc_bets repository failed to update a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryUpdateBet(ctx context.Context, bet storagemodels.BetCalculated) error {
+	updateBetSQL := "UPDATE bets SET selection_id=?, selection_coefficient=?, payment=? WHERE id=?"
+
+	statement, err := r.dbExecutor.PrepareContext(ctx, updateBetSQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = statement.ExecContext(ctx, bet.Payment, bet.Id)
+	return err
+}
+
+// GetBetCalculatedByID fetches a calculated bet from the database and returns it. The second returned value indicates
+// whether the bet exists in DB. If the bet does not exist, an error will not be returned.
+func (r *BetRepository) GetBetCalculatedByID(ctx context.Context, id string) (domainmodels.BetCalculated, bool, error) {
+	storageBet, err := r.queryGetBetByID(ctx, id)
+	if err == sql.ErrNoRows {
+		return domainmodels.BetCalculated{}, false, nil
+	}
+	if err != nil {
+		return domainmodels.BetCalculated{}, false, errors.Wrap(err, "calc_bets repository failed to get a bet with id "+id)
+	}
+
+	domainBet := r.betMapper.MapStorageBetToDomainBet(storageBet)
+	return domainBet, true, nil
+}
+
+func (r *BetRepository) queryGetBetByID(ctx context.Context, id string) (storagemodels.BetCalculated, error) {
+	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id='"+id+"';")
+	if err != nil {
+		return storagemodels.BetCalculated{}, err
+	}
+	defer row.Close()
+
+	// This will move to the "next" result (which is the only result, because a single bet is fetched).
+	row.Next()
+
+	var selectionId string
+	var selectionCoefficient int
+	var paymentSql sql.NullInt64
```

Comment:
Payment will always be set, it is not nullable in the DB (check data model in README or in `dbinitializer` script).